go/chat/sync.go

```diff
@@ -225,10 +225,16 @@ func (s *Syncer) Sync(ctx context.Context, cli chat1.RemoteInterface, uid gregor
 	return s.sync(ctx, cli, uid, syncRes)
 }
 
-func (s *Syncer) shouldDoFullReloadFromIncremental(convs []chat1.Conversation) bool {
+func (s *Syncer) shouldDoFullReloadFromIncremental(ctx context.Context, syncRes storage.InboxSyncRes,
+	convs []chat1.Conversation) bool {
+	if syncRes.TeamTypeChanged {
+		s.Debug(ctx, "shouldDoFullReloadFromIncremental: team type changed")
+		return true
+	}
 	for _, conv := range convs {
 		switch conv.ReaderInfo.Status {
 		case chat1.ConversationMemberStatus_LEFT, chat1.ConversationMemberStatus_REMOVED:
+			s.Debug(ctx, "shouldDoFullReloadFromIncremental: join or leave conv")
```

Comment:
leave or removed?

Reply:
basically just that the conversation should disappear from UI inbox list. `REMOVED` happens if you leave the team, `LEFT` happens if you explicitly leave the channel. I made a distinction there, so that if you re-join the team, you automatlically come back into `REMOVED` channels, but not `LEFT` channels.

Reply:
I was referring to the text in the log message.  It says "join or leave", I thought you might have meant "leave or removed".

Reply:
Ah yeah, I can fix that, sorry!